In [1]:
import pandas as pd

import warnings
warnings.simplefilter('ignore')

In [2]:
# Importing the Dataset

df = pd.read_csv(r'clean.csv')
df

Age  Heart_Rate  Calories  Gender_male
0       68       105.0     231.0            1
1       20        94.0      66.0            0
2       69        88.0      26.0            1
3       34       100.0      71.0            0
4       27        81.0      35.0            0
...    ...         ...       ...          ...
14995   20        92.0      45.0            0
14996   27        85.0      23.0            0
14997   43        90.0      75.0            0
14998   78        84.0      11.0            1
14999   63        92.0      98.0            1

[15000 rows x 4 columns]

In [3]:
# Seperating the Dependent and Independent variables.

X = df.drop(columns=['Calories'])
y = df['Calories']

In [4]:
# Splitting the data into Train and Test
from sklearn.model_selection import train_test_split

x_train,x_test,y_train,y_test = train_test_split(X,y,random_state=42,train_size=0.8)

In [7]:
# Apply HyperParameter Tunning on Ridge Regression

from sklearn.model_selection import GridSearchCV
from sklearn.linear_model import Ridge

estimator = Ridge()

param_grid = {'alpha':list(range(1,300))}

grid = GridSearchCV(estimator,param_grid,cv=5,scoring='r2')
grid.fit(x_train,y_train)
grid.best_params_

{'alpha': 299}

The alpha value is as below 

- Between range(1,100) - 99
- Between range(1,200) - 199
- Between range(1,300) - 299
- So that, we can choose the default alpha value of 1

In [8]:
# Build the Ridge Regression model with alpha = 1

model = Ridge(alpha=1)
model.fit(x_train,y_train)

print('Intercept is :',model.intercept_)
print('Coefficients are :',model.coef_)

Intercept is : -491.2610492184658
Coefficients are : [0.52525312 5.83770105 1.127168  ]


In [9]:
# Predicting on Train data
ypred_train = model.predict(x_train)

# Calculating the Train r2 score and Cross Validation Score
from sklearn.metrics import r2_score
from sklearn.model_selection import cross_val_score

print('Train r2 score is :',r2_score(y_train,ypred_train))
print('Cross validation score is :',cross_val_score(model,x_train,y_train,cv=5).mean())

# Predicting on Test data
ypred_test = model.predict(x_test)

# Calculating the Test r2 score
print('Test r2 score is :',r2_score(y_test,ypred_test))

Train r2 score is : 0.825934809010672
Cross validation score is : 0.8253414916888169
Test r2 score is : 0.8325080483280156
